src/Adapter/MSTest.CoreAdapter/Discovery/TestMethodValidator.cs

```diff
@@ -51,12 +51,15 @@ internal virtual bool IsValidTestMethod(MethodInfo testMethodInfo, Type type, IC
                 return false;
             }
 
+            var ignoreParameters = this.reflectHelper.DoesAttributeImplement(testMethodInfo, typeof(ITestDataSource), true);
+
             // Todo: Decide wheter parameter count matters.
             // The isGenericMethod check below id to verify that there are no closed generic methods slipping through.
             // Closed generic methods being GenericMethod<int> and open being GenericMethod<T>.
             var isValidTestMethod = testMethodInfo.IsPublic && !testMethodInfo.IsAbstract && !testMethodInfo.IsStatic
                                     && !testMethodInfo.IsGenericMethod
-                                    && testMethodInfo.IsVoidOrTaskReturnType();
+                                    && testMethodInfo.IsVoidOrTaskReturnType()
```

Comment:
What is the warning message users sees, should we add a test for that ?

Also, if parameters are more than what DataSource needs, what then ?

Reply:
Added test.

If there is parameter count mismatch, if will be caught while trying to invoke the test Method.